src/Turn.js

```diff
@@ -1,6 +1,70 @@
+const C = require('../src/constants.js')
+var clone = require('clone')
 class Turn {
-  constructor () {
-    console.log('Implement me!')
+  constructor (board, bikes, inputs) {
+    this.board = board
+    this.bikes = bikes
+    this.inputs = inputs
+  }
+  setInput (pos, input) {
+    this.inputs[pos] = input
+  }
+  evolve () {
+    var newBoard = clone(this.board)
+    var newBikes = clone(this.bikes)
+    var newInputs = []
+    for (var i = 0; i < this.bikes.length; i++) {
+      if (!(this.inputs[i] === null)) {
```

Comment:
Se podría simplificar el código declarando al principio del `for`:
```js
const bike = this.bikes[i]
const input = this.inputs[i]
```
y substituyendo en el código